data/schema/core/addons.cfg

```diff
@@ -0,0 +1,470 @@
+	# TODO: These are probably only allowed if type is mp or hybrid?
+	{DEFAULT_KEY min_players int 2}
+	{SIMPLE_KEY max_players int}
+	{DEFAULT_KEY allow_era_choice bool yes}
+	{DEFAULT_KEY require_campaign bool yes}
+[/tag]
+[tag]
+	name="era"
+	max=infinite
+	super="$addon"
+	{ADDON_DESCRIPTION}
+	{BASE_COMPATIBILITY_KEYS}
+	{SIMPLE_KEY force_modification string_list}
+	{SIMPLE_KEY require_era bool}
+	{DEFAULT_KEY hide_help bool no}
+	{DEFAULT_KEY type addon_type sp}
+	[link]
+		name="multiplayer_side"
+	[/link]
+	[tag]
+		name="options"
+		[tag]
+			name="checkbox"
+			max=infinite
+			{REQUIRED_KEY id string}
+			{DEFAULT_KEY default bool no}
+			{SIMPLE_KEY name t_string}
+			{SIMPLE_KEY description t_string}
+		[/tag]
+		[tag]
+			name="choice"
+			max=infinite
+			{REQUIRED_KEY id string}
+			{SIMPLE_KEY default string}
+			{SIMPLE_KEY name t_string}
+			{SIMPLE_KEY description t_string}
+			[tag]
+				name="item"
+				max=infinite
+				{REQUIRED_KEY value string}
+				{SIMPLE_KEY name t_string}
+			[/tag]
+		[/tag]
+		[tag]
+			name="entry"
+			max=infinite
+			{REQUIRED_KEY id string}
+			{SIMPLE_KEY default string}
+			{SIMPLE_KEY name t_string}
+			{SIMPLE_KEY description t_string}
+		[/tag]
+		[tag]
+			name="slider"
+			max=infinite
+			{REQUIRED_KEY id string}
+			{SIMPLE_KEY default int}
+			{SIMPLE_KEY min int}
+			{SIMPLE_KEY max int}
+			{SIMPLE_KEY step int}
+			{SIMPLE_KEY name t_string}
+			{SIMPLE_KEY description t_string}
+		[/tag]
+		[tag]
+			name="combo"
+			max=infinite
+			super="era/options/choice"
+			deprecated=yes
+		[/tag]
+	[/tag]
+[/tag]
+[tag]
+	name="modification"
+	max=infinite
+	super="$addon"
+	{ADDON_DESCRIPTION}
+	# TODO: Is this the correct default value?
+	{DEFAULT_KEY type addon_type mp}
+	{BASE_COMPATIBILITY_KEYS}
+	{SIMPLE_KEY ignore_incompatible_scenario string_list}
+	{DEFAULT_KEY require_modification bool no}
+	[link]
+		name="era/options"
+	[/link]
+[/tag]
+[tag]
+	name="resource"
+	max=infinite
+	super="$addon"
+[/tag]
+[tag]
+	name="scenario"
+	max=infinite
+	super="$addon"
+	{ADDON_DESCRIPTION}
+	{SIMPLE_KEY next_scenario string}
+	{SIMPLE_KEY name t_string}
+	{SIMPLE_KEY map_data string}
+	{DEFAULT_KEY turns int -1}
+	{DEFAULT_KEY turn_at int 1}
+	{DEFAULT_KEY random_start_time bool,int_list no} # Note: Is it random_start_time or random_starting_time? (There's some uses of the latter)
+	{DEPRECATED_KEY music string}
+	{SIMPLE_KEY defeat_music string}
+	{SIMPLE_KEY victory_music string}
+	{SIMPLE_KEY theme string_list}
+	{DEFAULT_KEY victory_when_enemies_defeated bool yes}
+	# TODO: The following two keys are only valid if the above key is yes (or omitted)
+	{SIMPLE_KEY carryover_percentage int}
+	{SIMPLE_KEY carryover_add bool}
+	{DEFAULT_KEY remove_from_carryover_on_defeat bool yes}
+	{DEFAULT_KEY disallow_recall bool no}
+	{DEFAULT_KEY experience_modifier int_percent 100}
+	{DEFAULT_KEY current_time int 0}
+	{SIMPLE_KEY map_generation string}
+	{SIMPLE_KEY scenario_generation string}
```

Comment:
only one of map_generation /scenario_generation  can be used not both at the same time.

Reply:
The schema currents is incapable of representing that restraint. Not sure if it's reply that important to be honest.